src/Hazelcast.Net.Testing/EnumerableExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Hazelcast.Testing
+{
+    public static class EnumerableExtensions
+    {
+        public static IEnumerable<IList<T>> Batch<T>(this IEnumerable<T> source, int batchSize)
```

Comment:
Can we have this return `IEnumerable<IEnumerable<T>>` and avoid allocating lists?

Reply:
In such case `List`s will be allocated implicitly, there's no way to split `IEnumerable` into independent sub`IEnumerable`s.
I don't optimization matters that much here, as this is for tests only.

Reply:
ah yes it's in the `Testing` project not the main project, my mistake.
and... doh - an `IEnumerable<IEnumerable<T>>` without lists requires that each inner enumerable be enumerated in order... ok let's keep it like this.